src/NUnitFramework/framework/Attributes/TheoryAttribute.cs

```diff
@@ -59,5 +62,22 @@ public class TheoryAttribute : CombiningStrategyAttribute, ITestBuilder, IImplyF
             new ParameterDataProvider(new DatapointProvider(), new ParameterDataSourceProvider()))
         {
         }
+
+        /// <summary>
+        /// Construct one or more TestMethods from a given MethodInfo,
+        /// using available parameter data.
+        /// </summary>
+        /// <param name="method">The MethodInfo for which tests are to be constructed.</param>
+        /// <param name="suite">The suite to which the tests will be added.</param>
+        /// <returns>One or more TestMethods</returns>
+        public override IEnumerable<TestMethod> BuildFrom(IMethodInfo method, Test suite)
+        {
+            var cases = base.BuildFrom(method, suite).ToList();
+
+            if (cases.Count == 0 && IsTreatedAsSuite(method))
+                throw new InvalidOperationException("There are no test cases");
```

Comment:
No problem. I wanted you to see this.
The next option I think is the simplest is `bool ITestFixture.RequireTestCases`. It's also much cleaner, no exception handling. I'll demonstrate that next.